File: 1331. Rank Transform of an Array/solution.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        n = len(arr)
        hm = {}
        nums = arr[:]

        nums.sort()
        rank = 1

        for num in nums:
            if num not in hm:
                hm[num] = rank
                rank += 1

        return [hm[num] for num in arr]

File: 1331. Rank Transform of an Array/solution.cpp
class Solution {
public:
    vector<int> arrayRankTransform(vector<int>& arr) {
        vector<int>nums(arr);
        unordered_map<int,int>index;
        sort(begin(nums),end(nums));
        int rank = 1 , n = arr.size();

        for(int i=0;i<n;i++){
            int val = nums[i];
            if(!index.count(val)){
                index[val] = rank;
                rank++;
            }   
        }

        for(int i=0;i<n;i++)
            nums[i] = index[arr[i]];
        
        return nums;
    }
};